modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -523,7 +535,7 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
         val ref = Type.Ref(shapeId.getNamespace(), shapeId.getName())
         val (name, node) = map.head // unions are encoded as objects
         val alt = union.alts.find(_.name == name).get
-        TypedNode.AltTN(ref, name, NodeAndType(node, alt.tpe))
+        TypedNode.AltTN(ref, name, NodeAndType(node, alt.tpe.toOption.get))
```

Comment:
I need to add some examples that hit this code path and sort out what to do here... Not exactly sure how to get around this other than changing the AltTN type a bit

Reply:
> other than changing the AltTN type 

Let's go ahead and do that